README.md

```diff
@@ -25,14 +25,14 @@ type Response struct {
 
 #### HTTP Layer
 
-Use the method `request.WithRequestID` to create and put a Request ID in context
+Use the method `request.WithID` to create and put a Request ID in context
```

Comment:
We have `GetIDFromContext` but `WithID` when getting the value. But we also have `SetInHTTPRequest` and `GetFromHTTPRequest`.
Maybe we should have a `SetIDInContext`.
Or if we wanna the With keyword `ContextWithID()` and `HTTPRequestWitthID`
And the get could be `request.FromContext()` and `request.FromHTTPReqeust()`

Reply:
Originally, the intention of using `WithFooBar` was to keep the same use style than the context method `With`. It was assumed to be more intuitive and easier to remember, since `With` would imply that a context is expected. Using `ContextWithFooBar`,or `foobar.WithContext` in zerolog style, are valid alternatives that lose a bit of the original intention. Zerolog alternative works a bit poorly in the `trace` package as right now there's both `ID` and `Trace` types to be used.

About dealing with `FromFoobar` and `SetInFooBar`, it could be normalized into `ToFooBar` and `FromFooBar`, keeping the same sense of movement of the `to` and `from` words. `To` is also common on casts, though.

I'd not try to group http helpers functions name with the context helper functions names. The intention is about the same, adding and retrieving data to stuff, but the differences on how they are perceived (as in read and thought around) would make a single wording choice always feel a little bit off, in my opinion.

Reply:
Regarding the proposed changes I did approve the PR.
I do not feel ready to suggest a standard. If changed I'll make a new review.